lib/blocs/FoodDetailPageBloc.dart

```diff
@@ -18,54 +18,54 @@ import 'dart:math';
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
-import 'package:food_delivery_app/models/Food.dart';
+import 'package:food_delivery_app/models/food_model.dart';
 import 'package:food_delivery_app/resourese/auth_methods.dart';
 import 'package:food_delivery_app/resourese/databaseSQL.dart';
 import 'package:food_delivery_app/resourese/firebase_helper.dart';
 
 class FoodDetailPageBloc with ChangeNotifier {
-  
   AuthMethods mAuthMethods = AuthMethods();
   FirebaseHelper mFirebaseHelper = FirebaseHelper();
 
-  List<Food> foodList=[];
+  List<FoodModel> foodList = [];
 
   // I dont implemented rating system,
   // so just for good UI, i am showing random value of rates from 0.00 to 5.00,
   // I been lazy here XD.
   var random = new Random();
-  String rating = "1.00"; 
+  String rating = "1.00";
 
   // no of items add to list
   int mItemCount = 1;
 
-  BuildContext context;
+  BuildContext? context;
 
-  addToCart(Food food) async{
-      DatabaseSql databaseSql=DatabaseSql();
-      await databaseSql.openDatabaseSql();
-      await databaseSql.insertData(food);
-      await databaseSql.getData();
-      final snackBar = SnackBar(
-        content: Text('Food Added To Cart'),
-        action: SnackBarAction(
-          label: 'Undo',
-          onPressed: () {
-            // Some code to undo the change.
-          },
-        ),
-      );
-      mItemCount = 1;
-      ScaffoldMessenger.of(context).showSnackBar(snackBar);
-      notifyListeners();
-      
+  addToCart(FoodModel food) async {
+    DatabaseSql databaseSql = DatabaseSql();
+    await databaseSql.openDatabaseSql();
+    await databaseSql.insertData(food);
+    await databaseSql.getData();
+    final snackBar = SnackBar(
+      content: Text('Food Added To Cart'),
+      action: SnackBarAction(
+        label: 'Undo',
+        onPressed: () {
+          // Some code to undo the change.
+        },
+      ),
+    );
+    mItemCount = 1;
+    if (context != null) {
+      ScaffoldMessenger.of(context!).showSnackBar(snackBar);
+    }
```

Comment:
I wonder why need "!" after checking not null value in the if statement

Reply:
for some cases even if we null check It shows as an error in ide like 'The argument type 'BuildContext?' can't be assigned to the parameter type 'BuildContext" '.  Maybe it's some Dart's static analysis tools bug. I am using vscode.